ddtrace/appsec/processor.py

```diff
@@ -190,6 +192,57 @@ def _update_rules(self, new_rules):
         except TypeError:
             log.debug("Error updating ASM rules", exc_info=True)
 
+    def _set_metrics(self):
```

Comment:
Ideally this PR should introduce the functionality to queue and send telemetry metrics. Then in a follow up PR we can use the telemetry writer to send metrics in different products. Can we decouple adding the telemetry metrics writer change from sending telemetry metrics in ASM products? 

This way if there is an issue with the telemetry writer (performance or a bug) we can just revert/fix the PRs which send telemetry and not all the logic in this PR. This is just a suggestion, if we need this functionality merged ASAP then I am fine with leaving this PR as is

Reply:
Thanks! your suggestion is a good approach! If I start now to split the PR in smaller PRs I'll became mad :sweat_smile: What do you think about this idea?
- I'm continue in Telemetry Metrics in this branch
- When we're ready for the final review and merge I can split this PR into "Telemetry Metrics Core" and "ASM Telemetry Metrics"

is it OK for you? I'd love to know if this approach is quite so smooth for all of us :+1:

Reply:
Yup that sounds great to me, it would be nice to get @Yun-Kim feedback on the performance benchmarks. Thanks for sticking with this. This PR looks good to me. If you can get another thumbs up from another review then we can split up and merging this change.

Reply:
I'm curious why this change results in a regression in the `span-add-metrics` microbenchmark scenario, which specifically only call `span.set_metrics()` and shouldn't be impacted by telemetry, unless I'm missing something. @avara1986 do you have any clues as to why we're seeing this?

Reply:
ey Yun! no, I haven't. It's strange because the microbenchmark didn't report anything in the last commits, even after adding specific benchmark tests for Telemetry Metrics. `span-add-metrics` reports a regression after a merge of 1.x in this branch. But I merged again 1.x into this branch and the regression alert disappear. Flaky? I figure out that to be sure :+1: